Pidaq/Lib/can_manager/can_manager.py

```diff
@@ -0,0 +1,169 @@
+                    reading['reading'],
+                    reading['conversion_factor'],
+                    reading['conversion_factor_type']
+                )
+                self.message_ids.append(int(reading['message_id'], 16))
+
+    def send_message(self, id: int, data: list) -> None:
+        if id in self.message_ids:
+            raise Exception(f'Error: ID: {id} is already in use')
+        message = can.Message(arbitration_id=id, data=data)
+        self.bus.send(message)
+
+    def read_bus(self) -> can.Message:
+        message = self.bus.recv()
+        return message
+
+    def assign_message_data(self, bus_message: can.Message) -> None:
+        """Assigns message data to the correct SensorReading object
+
+        Assigns message data to the correct SensorReading object
+        based on the arbitration_id of the message, checks to ensure
+        that arbitration_id is in list of ids for the test
+
+        Parameters:
+            bus_message(can.Message)
+        """
+        message_id = bus_message.arbitration_id
+        if message_id in self.message_ids:
+            self.messages[message_id].data = bytes(bus_message.data)
+
+
+class SensorReading:
+    """Class used to store information about sensor readings to be received on can bus
+
+    This class contains all the information about the project-specific sensor measurement
+    readings that will be sent over the can bus from the sensor board to the raspberry pi
+
+    Attributes:
+        message_id (int): represents the can arbitration id, is a hexidecimal integer
+        reading (str): contains the name of the measurement
+        conversion_factor (int or float)
+        conversion_factor_type (str)
+        data (int or float)
+
+    Methods:
+
+    """
+    def __init__(self, project: str, message_id: str, reading: str,
+                 conversion_factor: str, conversion_factor_type: str) -> None:
+        self.message_id = int(message_id, 16)
+        self.reading = reading
+        self.conversion_factor = parse_conversion_factor(conversion_factor,
+                                                         conversion_factor_type)
+        self.data = None
+
+
+def parse_conversion_factor(conversion_factor: str, conversion_factor_type: str):
+    """Converts a string conversion factor into the correct numerical representation
+
+    Given the string representation of a conversion factor, as well as its type,
+    this function returns the conversion factor in its correct numerical representation,
+    either int or float
+
+    Parameters:
+        conversion_factor(str)
+        conversion_factor_type(str)
+
+    Returns:
+        conversion_factor(int or float)
+    """
+    if conversion_factor_type.lower() == 'float':
+        return float(conversion_factor)
+    elif conversion_factor_type.lower() == 'int':
+        return int(conversion_factor)
+    else:
+        raise ValueError('Error: conversion_factor_type is invalid')
```

Comment:
We may need to rethink the way we perform calculations on some of our readings (maybe with more specific child classes). We can discuss this at the working session this weekend. This way would work well for ADC readings. We could also consider using one of the message bytes to indicate the resolution of the reading - this would be a little more dynamic.

Reply:
Yup I agree that is a good idea, we'll discuss this weekend